File: pathogens to antibiotics/matrix_convert.ts
import fs from 'fs';
import parse from 'csv-parse/lib/sync';

// Define the path to the CSV file
const csvFilePath = 'path_to_your_csv_file.csv';

// Read the CSV file content
const csvContent = fs.readFileSync(csvFilePath);

// Parse the CSV content
const records = parse(csvContent, {
  skip_empty_lines: true,
});

// Extract the pathogen list headers (skip the first header row)
const [, ...pathogenHeaders] = records[1]; // This contains the actual pathogen list names

// Initialize the dictionary with pathogen list names as keys
const pathogenAntibiotics: { [key: string]: string[] } = {};
pathogenHeaders.forEach(header => {
  pathogenAntibiotics[header] = [];
});

// Read the remaining rows in the CSV file
records.slice(2).forEach((row: string[]) => {
  // The first column in each row is the antibiotic name
  const antibioticName = row[0];

  // Loop through each cell in the row after the antibiotic name
  row.slice(1).forEach((cell, index) => {
    // Check if the cell has an 'x', indicating an association
    if (cell.toLowerCase() === 'x') {
      // Add the antibiotic to the appropriate pathogen list in the dictionary
      const pathogenList = pathogenHeaders[index];
      pathogenAntibiotics[pathogenList].push(antibioticName);
    }
  });
});

// Convert the dictionary to a JSON object
const jsonObject = JSON.stringify(pathogenAntibiotics, null, 4);

// Output the JSON object
console.log(jsonObject);

File: pathogens to antibiotics/matrix_convert.py
import csv
import json

# Define the path to the CSV file
# csv_file_path = 'CLSI_Matrix.csv'
csv_file_path = 'EUCAST_Matrix.csv'

# Initialize an empty dictionary to hold the pathogen lists and their associated antibiotics
pathogen_antibiotics = {}

# Open the CSV file for reading
with open(csv_file_path, mode='r') as csvfile:
    # Create a CSV reader object
    csvreader = csv.reader(csvfile)
    
    # Extract the pathogen list headers (skip the first header row)
    headers = next(csvreader)[1:]  # Skip the first empty header
    pathogen_headers = next(csvreader)[1:]  # This contains the actual pathogen list names
    
    # Initialize the dictionary with pathogen list names as keys
    for header in pathogen_headers:
        pathogen_antibiotics[header] = []
    
    # Read the remaining rows in the CSV file
    for row in csvreader:
        # The first column in each row is the antibiotic name
        antibiotic_name = row[0]
        
        # Loop through each cell in the row after the antibiotic name
        for index, cell in enumerate(row[1:], start=1):  # Offset by 1 to align with the header index
            # Check if the cell has an 'x', indicating an association
            if cell.lower() == 'x':
                # Add the antibiotic to the appropriate pathogen list in the dictionary
                pathogen_list = pathogen_headers[index-1]  # Adjust index to match header list
                pathogen_antibiotics[pathogen_list].append(antibiotic_name)

# Convert the dictionary to a JSON object
json_object = json.dumps(pathogen_antibiotics, indent=4)

# Output the JSON object
print(json_object)
